pkg/sql/opt/metadata.go

```diff
@@ -415,6 +379,87 @@ func (md *Metadata) AddTable(tab cat.Table, alias *tree.TableName) TableID {
 	return tabID
 }
 
+// DuplicateTable creates a new reference to the table with the given ID. All
+// columns are added to the metadata with new column IDs. If mutation columns
+// are present, they are added after active columns. The ID of the new table
+// reference is returned. This function panics if a table with the given ID does
+// not exists in the metadata.
+//
+// remapColumnIDs must be a function that remaps the column IDs within a
+// ScalarExpr to new column IDs. It takes as arguments a ScalarExpr and a
+// mapping of old column IDs to new column IDs, and returns a new ScalarExpr.
+// This function is used when duplicating Constraints, ComputedCols, and
+// PartialIndexPredicates.
```

Comment:
Maybe a comment explaining why it takes a callback function rather than directly duplicating the expressions.

Reply:
Done.